### PR TITLE
GraphQl Schema actualization with latest released changes.

### PR DESCRIPTION
**Pull request by swnsma**
## Problem
1. In-Store Pickup GraphQl schema is not up-to-date with released and [documented](https://devdocs.magento.com/guides/v2.4/graphql/queries/pickup-locations.html) schema.
2. Products assignment intersection filter was missed from GraphQl API in the initial implementation.

## Solution
1. Update schema from the inventory repo.
2. Introduce `productsInfo` parameter for assignment intersection filter.

## Requested Reviewers
@joni-jones
@paliarush

## Additional
GraphQl Implementation in: https://github.com/magento/inventory/pull/3161
Original ticket for functionality: https://github.com/magento/inventory/issues/2825
REST API implementation: https://github.com/magento/inventory/pull/2867

### CODE REVIEW

File: design-documents/graph-ql/coverage/store-pickup.graphqls [hunk 1]
```diff
@@ -1,12 +1,15 @@
 type Query {
     pickupLocations (
-        filter: PickupLocationFilterInput,
+        area: AreaInput,
+        filters: PickupLocationFilterInput,
+        sort: PickupLocationSortInput,
         pageSize: Int = 20,
-        currentPage: Int = 1
+        currentPage: Int = 1,
+        productsInfo: [ProductInfoInput]
```
Comment (melnikovi) [inline design-documents/graph-ql/coverage/store-pickup.graphqls:8]: Does this field exist?

Comment (swnsma) [reply design-documents/graph-ql/coverage/store-pickup.graphqls:8]: It is going to be delivered in https://github.com/magento/inventory/pull/3161

**Comment by melnikovi** [comment]
Please resolve merge conflicts.

**Comment by swnsma** [comment]
@melnikovi resolved

**Pull request merged**